### PR TITLE
Fix: tupled app args with transaction type

### PR DESCRIPTION
**Pull request by barnjamin**
addresses #576

### CODE REVIEW

**Review by tonyfloatersu — COMMENTED** [review]
Hey, it's also Hang, please consider add a line in CHANGELOG

File: pyteal/ast/router.py [hunk 1]
```diff
@@ -431,13 +432,10 @@ def wrap_handler(
             # de-tuple into specific values using `store_into` on
             # each element of the tuple'd arguments
             if tuplify:
```
Comment (ahangsu) [inline pyteal/ast/router.py:434]:
```suggestion
            if tuplify:
                tupled_app_args = app_arg_vals[METHOD_ARG_NUM_CUTOFF - 1 :]
```

File: pyteal/ast/router.py [hunk 2]
```diff
@@ -361,6 +361,11 @@ def wrap_handler(
                 ats for ats in arg_vals if not isinstance(ats, abi.Transaction)
             ]
 
+            # only transaction args (these are omitted from app args)
+            txn_arg_vals: list[abi.Transaction] = [
+                ats for ats in arg_vals if isinstance(ats, abi.Transaction)
+            ]
+
```
Comment (ahangsu) [inline pyteal/ast/router.py:368]:
```suggestion
```

File: pyteal/ast/router.py [hunk 3]
```diff
@@ -373,16 +378,12 @@ def wrap_handler(
             # assign to a var here since we modify app_arg_vals later
             tuplify = len(app_arg_vals) > METHOD_ARG_NUM_CUTOFF
```
Comment (ahangsu) [inline pyteal/ast/router.py:380]:
```suggestion
            tuplify = len(app_arg_vals) > METHOD_ARG_NUM_CUTOFF

            # only transaction args (these are omitted from app args)
            txn_arg_vals: list[abi.Transaction] = [
                ats for ats in arg_vals if isinstance(ats, abi.Transaction)
            ]

```

**Review by ahangsu — COMMENTED** [review]
minor comments to minimize the change, the move at this point seems reasonable to me.

Good to have the issue reproducing code in test case, with transaction being the last arg.

**Comment by ahangsu** [comment]
oh snap something goes wild, looking into it

File: pyteal/ast/router.py [hunk 4]
```diff
@@ -383,6 +383,7 @@ def wrap_handler(
                 last_arg_specs_grouped: list[abi.TypeSpec] = [
                     t.type_spec() for t in app_arg_vals[METHOD_ARG_NUM_CUTOFF - 1 :]
                 ]
+                tupled_app_args = app_arg_vals[METHOD_ARG_NUM_CUTOFF - 1 :]
```
Comment (ahangsu) [inline pyteal/ast/router.py:386]:
```suggestion
                tupled_app_args = app_arg_vals[METHOD_ARG_NUM_CUTOFF - 1 :]
                last_arg_specs_grouped: list[abi.TypeSpec] = [
                    t.type_spec() for t in tupled_app_args
                ]
```

**Review by ahangsu — APPROVED** [review]
I think it is in a good shape, also requesting a pass from @jasonpaulos

**Review by michaeldiamant — APPROVED** [review]
@barnjamin Thanks for the fix - looks correct to me.  

Can address out-of-band to the PR:  I'd feel more confident if we can generate arbitrary test examples capturing all `BaseType`s.

**Review by jasonpaulos — APPROVED** [review]
Looks correct, thanks for fixing this

**Pull request merged**